src/Stripe.net/Services/PaymentIntents/PaymentIntentUpdateOptions.cs

```diff
@@ -1,9 +1,125 @@
+
+        /// <summary>
+        /// The list of payment method types (e.g. card) that this PaymentIntent is allowed to use.
+        /// </summary>
+        [JsonProperty("payment_method_types")]
+        public List<string> PaymentMethodTypes { get; set; }
+
+        /// <summary>
+        /// Email address that the receipt for the resulting payment will be sent to.
+        /// </summary>
+        [JsonProperty("receipt_email")]
+        public string ReceiptEmail { get; set; }
+
+        /// <summary>
+        /// If the PaymentIntent has a payment_method and a customer or if you’re attaching a
+        /// payment method to the PaymentIntent in this request, you can pass this as <c>true</c>
+        /// to save the payment method to the customer. Defaults to <c>false</c>.
+        /// If the payment method is already saved to a customer, this does nothing. If this type
+        /// of payment method cannot be saved to a customer, the request will error.
+        /// </summary>
+        [JsonProperty("save_payment_method")]
+        public bool? SavePaymentMethod { get; set; }
+
+        /// <summary>
+        /// Shipping information for this PaymentIntent.
+        /// </summary>
+        [JsonProperty("shipping")]
+        public ChargeShippingOptions Shipping { get; set; }
+
+        /// <summary>
+        /// ID of the Source to use with this PaymentIntent. Prefer using <see cref="PaymentMethodId"/>.
+        /// </summary>
+        [JsonProperty("source")]
+        public string SourceId { get; set; }
+
+        /// <summary>
+        /// Extra information about a PaymentIntent. This will appear on your customer’s statement
+        /// when this PaymentIntent succeeds in creating a charge.
+        /// </summary>
+        [JsonProperty("statement_descriptor")]
+        public string StatementDescriptor { get; set; }
+
+        /// <summary>
+        /// The data with which to automatically create a Transfer when the payment is finalized.
+        /// See the PaymentIntents <a href="https://stripe.com/docs/payments/payment-intents/use-cases#connected-accounts">
+        /// use case for connected accounts</a> for details.
+        /// </summary>
+        [JsonProperty("transfer_data")]
+        public PaymentIntentTransferDataOptions TransferData { get; set; }
+
+        /// <summary>
+        /// A string that identifies the resulting payment as part of a group. See the
+        /// PaymentIntents <a href="https://stripe.com/docs/payments/payment-intents/use-cases#connected-accounts">
+        /// use case for connected accounts</a> for details.
+        /// </summary>
+        [JsonProperty("transfer_group")]
+        public string TransferGroup { get; set; }
+
+        [Obsolete("Use PaymentMethodTypes")]
+        [JsonProperty("allowed_source_types")]
+        public List<string> AllowedSourceTypes { get; set; }
+
+        [Obsolete("This field was never supported in the API so marking as deprecated/")]
```

Comment:
If this was indeed never supported by the API, I think we can simply remove the field without a major version.

Reply:
Yeah but it will likely be supported in a few weeks so I wanted to avoid the churn :p

Reply:
We'd still have to remove the `Obsolete` attribute, but fair enough :)

Can you change the comment to just "This field was never supported in the API" (or maybe "This field is not yet supported in the API")?

Reply:
Haha, too slow I guess. Not a big deal.